compiler/noirc_frontend/src/hir/comptime/interpreter/foreign.rs

```diff
@@ -76,25 +324,118 @@ fn poseidon2_permutation(
 fn keccakf1600(
     interner: &mut NodeInterner,
     arguments: Vec<(Value, Location)>,
     location: Location,
 ) -> IResult<Value> {
     let input = check_one_argument(arguments, location)?;
-    let input_location = input.1;
-
-    let (input, typ) = get_array(interner, input)?;
 
-    let input = try_vecmap(input, |integer| get_u64((integer, input_location)))?;
+    let (state, typ) = get_fixed_array_map(interner, input, get_u64)?;
 
-    let mut state = [0u64; 25];
-    for (it, input_value) in state.iter_mut().zip(input.iter()) {
-        *it = *input_value;
-    }
     let result_lanes = acvm::blackbox_solver::keccakf1600(state)
         .map_err(|error| InterpreterError::BlackBoxError(error, location))?;
 
     let array: Vector<Value> = result_lanes.into_iter().map(Value::U64).collect();
     Ok(Value::Array(array, typ))
 }
+
+/// `pub fn sha256_compression(input: [u32; 16], state: [u32; 8]) -> [u32; 8]`
+fn sha256_compression(
+    interner: &mut NodeInterner,
+    arguments: Vec<(Value, Location)>,
+    location: Location,
+) -> IResult<Value> {
+    let (input, state) = check_two_arguments(arguments, location)?;
+
+    let (input, _) = get_fixed_array_map(interner, input, get_u32)?;
+    let (mut state, typ) = get_fixed_array_map(interner, state, get_u32)?;
+
+    acvm::blackbox_solver::sha256_compression(&mut state, &input);
+
+    let state = state.into_iter().map(Value::U32).collect();
+    Ok(Value::Array(state, typ))
+}
+
+/// Decode an `EmbeddedCurvePoint` struct.
+///
+/// Returns `(x, y, is_infinite)`.
+fn get_embedded_curve_point(
+    (value, location): (Value, Location),
+) -> IResult<(FieldElement, FieldElement, bool)> {
+    let (fields, typ) = get_struct_fields("EmbeddedCurvePoint", (value, location))?;
+    let x = get_struct_field("x", &fields, &typ, location, get_field)?;
+    let y = get_struct_field("y", &fields, &typ, location, get_field)?;
+    let is_infinite = get_struct_field("is_infinite", &fields, &typ, location, get_bool)?;
+    Ok((x, y, is_infinite))
+}
+
+/// Decode an `EmbeddedCurveScalar` struct.
+///
+/// Returns `(hi, lo)`.
+fn get_embedded_curve_scalar(
+    (value, location): (Value, Location),
+) -> IResult<(FieldElement, FieldElement)> {
+    let (fields, typ) = get_struct_fields("EmbeddedCurveScalar", (value, location))?;
+    let lo = get_struct_field("lo", &fields, &typ, location, get_field)?;
+    let hi = get_struct_field("hi", &fields, &typ, location, get_field)?;
+    Ok((lo, hi))
+}
+
+#[cfg(test)]
+mod tests {
+    use acvm::acir::BlackBoxFunc;
+    use noirc_errors::Location;
+    use strum::IntoEnumIterator;
+
+    use crate::hir::comptime::tests::with_interpreter;
+    use crate::hir::comptime::InterpreterError::{
+        ArgumentCountMismatch, Unimplemented, WillNotImplement,
+    };
+
+    use super::call_foreign;
+
+    /// Check that all `BlackBoxFunc` are covered by `call_foreign`.
+    #[test]
+    fn test_blackbox_implemented() {
```

Comment:
I'm not sure this is necessary tbh. If we want to keep this however then we should remove the `enum_iter` feature flag and just always have it on.

Reply:
I think it's nice to have one, if we care about providing nice error messages, might as well force ourselves to do so. 

I checked and found I think one instance where the enum iter was always on, not just in tests, all the others were conditional. It's a bit of a dance to have it and use it in tests, and I'm sure it doesn't help compile times to have it two ways, so I'm in favour of enabling it always.

Reply:
Removed the feature flag and enabled `EnumIter` for good.